source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -133,14 +133,15 @@ static int heap_alloc_internal_check(int16_t alloc_size)
 
     if (alloc_size) {
         //
-        temp_sector = alloc_size;
-        temp_int = (temp_sector % temp_size);
-        if (temp_int) {
-            temp_sector += (temp_size - temp_int);
-        }
+        temp_sector = (alloc_size+temp_size-1)/temp_size;
+//        temp_sector = alloc_size;
+//        temp_int = (temp_sector % temp_size);
+//        if (temp_int) {
+//            temp_sector += (temp_size - temp_int);
+//        }
 
 
-        temp_sector /= temp_size;
+//        temp_sector /= temp_size;
 
         temp_int = (heap_size / temp_size);
```

Comment:
Can you tidy this too? It's basically "if size > (heap_size - 2 words), not valid". Could do that at the top of the function in bytes next to the other size check, rather than in words.